text_extensions_for_pandas/array/tensor.py

```diff
@@ -70,6 +70,11 @@ def construct_array_type(cls):
         """
         return TensorArray
 
+    def __from_arrow__(self, extension_array):
+        from text_extensions_for_pandas.array.arrow_compat import ArrowTensorArray
```

Comment:
I put this in a new module because I'm not sure if pyarrow is a hard dependency on pandas, we can move it if you like

Reply:
Not having a hard dependency on Arrow makes sense, at least for now. We may want to revisit that once we get around to having a (presumably Arrow-backed) Pandas extension type for variable-length tensors.